sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/CancellationToken.java

```diff
@@ -2,20 +2,28 @@
 // Licensed under the MIT License.
 package com.azure.cosmos.implementation.changefeed;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+
 /**
  * Propagates notification that operations should be canceled..
  */
 public class CancellationToken {
     private final CancellationTokenSource tokenSource;
+    private AtomicBoolean cancellationRequested;
 
     public CancellationToken(CancellationTokenSource source) {
         this.tokenSource = source;
+        this.cancellationRequested = new AtomicBoolean(false);
+    }
+
+    public void cancel() {
```

Comment:
This is not needed, please remove; the cancelation of a token should only be done through the CancellationTokenSource.

Reply:
Chatted offline, will removed this and use cancellationTokenSource to cancel.